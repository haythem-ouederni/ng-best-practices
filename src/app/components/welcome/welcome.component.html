<div style="text-align:center">
  <h1
    [translate]="'app.env_message'"
    [translateParams]="{env: env}">
  </h1>
  <h1>
    {{ 'app.title' | translate: {title: title} }}
  </h1>
</div>

<button (click)="goToPageOne()" type="button" class="btn btn-primary">
  {{ 'app.buttons.go_to_page_one' |Â translate }}
</button>