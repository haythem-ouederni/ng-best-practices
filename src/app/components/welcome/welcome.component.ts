import {ROUTES_PATHS} from '@abpe/connection/lib/modules';
import {Component} from '@angular/core';
import {environment} from '../../../environments/environment';

@Component({
  selector: 'app-welcome',
  templateUrl: './welcome.component.html',
  styleUrls: ['./welcome.component.scss'],
})
export class WelcomeComponent {
  linkToPageOne = `/${environment.connection.basePath}/${ROUTES_PATHS.pageOne}`;

  title = 'Angular Best Practices Example';

  env = environment.production;
}
