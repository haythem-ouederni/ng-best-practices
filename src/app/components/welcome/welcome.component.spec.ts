import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {RouterTestingModule} from '@angular/router/testing';
import {WelcomeComponent} from './welcome.component';

describe('WelcomeComponent', () => {
  let component: WelcomeComponent;
  let fixture: ComponentFixture<WelcomeComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [WelcomeComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(WelcomeComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
