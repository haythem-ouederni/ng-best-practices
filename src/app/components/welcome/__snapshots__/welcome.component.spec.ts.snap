// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeComponent should match snapshot 1`] = `
<app-welcome
  env="false"
  linkToPageOne={[Function String]}
  title={[Function String]}
>
  <div
    style="text-align:center"
  >
    <h1>
      Production environment : false
    </h1>
    <h1>
       Welcome to Angular Best Practices Example! 
    </h1>
  </div><a
    href="/connection/page-one"
    ng-reflect-router-link="/connection/page-one"
  >
    Go to page one
  </a>
</app-welcome>
`;
